src/PostgREST/Error.hs

```diff
@@ -161,46 +192,52 @@ instance JSON.ToJSON PgError where
 
 instance JSON.ToJSON P.UsageError where
   toJSON (P.ConnectionError e) = JSON.object [
-    "code"    .= ("" :: Text),
+    "code"    .= ("PGCON" :: Text),
     "message" .= ("Database connection error. Retrying the connection." :: Text),
-    "details" .= (toSL $ fromMaybe "" e :: Text)]
+    "details" .= (toSL $ fromMaybe "" e :: Text),
+    "hint"    .= JSON.Null]
   toJSON (P.SessionError e) = JSON.toJSON e -- H.Error
 
 instance JSON.ToJSON H.QueryError where
   toJSON (H.QueryError _ _ e) = JSON.toJSON e
 
 instance JSON.ToJSON H.CommandError where
-  toJSON (H.ResultError (H.ServerError c m d h)) = case toS c of
-    'P':'T':_ -> JSON.object [
-        "details" .= (fmap toS d :: Maybe Text),
-        "hint"    .= (fmap toS h :: Maybe Text)]
-
-    _         -> JSON.object [
+  toJSON (H.ResultError (H.ServerError c m d h)) = JSON.object [
         "code"    .= (toS c      :: Text),
         "message" .= (toS m      :: Text),
         "details" .= (fmap toS d :: Maybe Text),
         "hint"    .= (fmap toS h :: Maybe Text)]
 
   toJSON (H.ResultError (H.UnexpectedResult m)) = JSON.object [
-    "message" .= (m :: Text)]
+    "code"    .= ("PGUNX" :: Text),
```

Comment:
This is more of a internal error - it'd happen if we mess a query. Hm, not sure if it'd valuable to have a special code for it.